base_layer/wallet/src/transaction_service/storage/models.rs

```diff
@@ -578,6 +578,30 @@ impl WalletTransaction {
             WalletTransaction::Completed(tx) => Some(tx.source_address.clone()),
         }
     }
+
+    pub fn is_pending(&self) -> bool {
+        match self {
+            WalletTransaction::PendingInbound(_) => true,
+            WalletTransaction::PendingOutbound(_) => true,
+            WalletTransaction::Completed(_) => false,
+        }
+    }
+
+    pub fn is_mined(&self) -> bool {
+        match self {
+            WalletTransaction::PendingInbound(_) => false,
+            WalletTransaction::PendingOutbound(_) => false,
+            WalletTransaction::Completed(tx) => tx.status.is_mined(),
+        }
+    }
```

Comment:
I think we should trust the actual status in the transaction
```suggestion
    pub fn is_mined(&self) -> bool {
        match self {
            WalletTransaction::PendingInbound(tx) => tx.status.is_mined(),
            WalletTransaction::PendingOutbound(tx) => tx.status.is_mined(),
            WalletTransaction::Completed(tx) => tx.status.is_mined(),
        }
    }
```

Reply:
Nope, 
pending tx CANNOT be mined, its not complete